ci/tests/watcher-master.yml

```diff
@@ -24,10 +24,42 @@ run_tempest: false
 cifmw_test_operator_concurrency: 1
 cifmw_test_operator_tempest_include_list: |
   watcher_tempest_plugin.*
-# We need to exclude client_functional tests until we have watcherclient installed in the
-# tempest container.
 # Some strategies execution tests are failing. Excluding until the work on the watcher-tempest-plugin
 # is finished upstream.
+# TODO(chandankumar): Drop watcher_tempest_plugin.*client_functional.* from here and also remove it
+# from watcher-tempest-plugin by openstack release 2024.2 eol.
 cifmw_test_operator_tempest_exclude_list: |
   watcher_tempest_plugin.*client_functional.*
   watcher_tempest_plugin.tests.scenario.test_execute_strategies.TestExecuteStrategies.test_execute_storage_capacity_balance_strategy
+
+# Tempest images cases
+# content_provider_os_registry_url is not null, It means an opendev depends-on is used
+# in the change. In this case, tempest image url should be content_provider_os_registry_url
+# tempest all container should be used for master branch.
+#
+# When content_provider_os_registry_url is null, in that case tempest image will be pulled
+# from quay.io/podified-master-centos9 for master branch.
+#
+# For antelope branch, In all cases, we will use tempest container from quay.io/podified-antelope-centos9
+# namespace and install watcher-tempest-plugin from master branch.
+
+cifmw_test_operator_tempest_registry: >-
```

Comment:
@raukadah right now this file is used in the `periodic-watcher-operator-validation-master` job ( the job that runs in the RDO master pipeline). I added it to reuse the tempest config and include/exclude lists, so adding the registry, namespace and tag will cause that job to use wrong content. Maybe we should move the shared tempest config to a different file and leave the new parameters here? That way we can drop this file from the periodic-watcher-operator-validation-master job and use the new one

Reply:
another option might be to change the order of the `cifmw_extras` files in the periodic job to ensure https://github.com/openstack-k8s-operators/ci-framework/blob/1d04e9f63439c09512c5a5f9333bd25e99598328/zuul.d/edpm_periodic.yaml#L96 comes last to have the right content, although I would need to test that to be sure

Reply:
Thank you @cescgina for the review. It will break periodic job. Let me fix it.

In `periodic-watcher-operator-validation-master` job, we have `watcher_dlrn_tag` and `watcher_registry_url` which set proper content for periodic line. I can add one more conditional using watcher_dlrn_tag and watcher_registry_url to set proper tempest vars in watcher-master.yml file.

Reply:
Testing here: https://review.rdoproject.org/r/c/testproject/+/57195

Reply:
thanks! looking at the testproject results it seems that we need to check if `content_provider_os_registry_url` is defined, because in the periodic job is not defined at all https://softwarefactory-project.io/zuul/t/rdoproject.org/build/ba9e21cd17044888ae8a0f62798643ad/log/job-output.txt#4102

Reply:
https://logserver.rdoproject.org/95/57195/2/check/periodic-watcher-operator-validation-master/a7a5f39/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/tempest-tests-tempest/tempest-tests-tempest-describe
```
Warning  Failed          64m (x4 over 66m)    kubelet            Failed to pull image "quay.rdoproject.org/podified-master-centos9/openstack-tempest-all:podified-ci-testing": reading manifest podified-ci-testing in quay.rdoproject.org/podified-master-centos9/openstack-tempest-all: manifest unknown
```
there is no podified-ci-testing tag here https://quay.rdoproject.org/repository/podified-master-centos9/openstack-tempest-all?tab=tags
let use cifmw_repo_setup_full_hash to get proper tag.

Reply:
@cescgina Based on https://logserver.rdoproject.org/95/57195/3/check/periodic-watcher-operator-validation-master/a3827d8/job-output.txt
```
containerImage: quay.rdoproject.org/podified-master-centos9/openstack-tempest-all:b9be57d8274b915b6eb7724d26629e8b
```
and dlrn md5 hash used  https://logserver.rdoproject.org/95/57195/3/check/periodic-watcher-operator-validation-master/a3827d8/controller/ci-framework-data/artifacts/repositories/delorean.repo.md5
```
b9be57d8274b915b6eb7724d26629e8b
```
Now it is working as expected.

Reply:
yep, looks good now, thanks @raukadah!